src/Constraint/MultiConstraint.php

```diff
@@ -117,4 +123,84 @@ public function __toString()
 
         return '[' . implode($this->conjunctive ? ' ' : ' || ', $constraints) . ']';
     }
+    /**
+     * @inheritDoc
+     */
+    public function getLowerBound()
+    {
+        $this->extractBounds();
+
+        return $this->lowerBound;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getUpperBound()
+    {
+        $this->extractBounds();
+
+        return $this->upperBound;
+    }
+
+    private function extractBounds()
+    {
+        if (null !== $this->lowerBound) {
+            return;
+        }
+
+        foreach ($this->constraints as $constraint) {
+            if (!$constraint instanceof BoundsProvidingInterface) {
+                continue; // TODO: or exception? Is it better to ignore those or not?
```

Comment:
If we just make this part of all constraints there is no need for this?

Reply:
Yes, if we bump to v2 there's no need for that.

Reply:
Do we need to go to v2 to add this method to the interface? Do we really guarantee that external implementations of this interface remain compatible?

Reply:
Well, adding a method to the interface is a BC break. I'm not sure what your policy here really is but this library is used by many (https://packagist.org/packages/composer/semver/dependents?order_by=downloads). I have used it standalone as well (never added my own constraint though).